File: objective-c/StoreKit/StoreKit_v2/StoreKit_v2/Templates/AButtonTemplate.m
//
//  AButtonTemplate.m
//  StoreKit_v1
//
//  Created by Alexandr Kirilov on 05/12/2022.
//

#import "AButtonTemplate.h"

@implementation AButtonTemplate

-(instancetype) initWithFrame:(CGRect)inFrame {
	
	self = [super initWithFrame:inFrame];
	if (self) {
		
		[self setBackgroundColor:__COLOR_PURPLE_DARK];
		[self setTranslatesAutoresizingMaskIntoConstraints:NO];
		
		[self.titleLabel setFont:[UIFont boldSystemFontOfSize:20.0]];
		[self.titleLabel setTextColor:__COLOR_WHITE];
		
	}
	return self;
}

-(void) setHighlighted:(BOOL)inHighlighted {
	
	if(inHighlighted) {
		self.backgroundColor = __COLOR_ORANGE_DARK;
	} else {
		self.backgroundColor = __COLOR_PURPLE_DARK;
	}
	[super setHighlighted:inHighlighted];
}

@end

File: swift/StoreKit/StoreKit2_v1/StoreKit2_v1/Templates/AButtonTemplate.swift
//
//  AButtonTemplate.swift
//  StoreKit2_v1
//
//  Created by Alexandr Kirilov on 19/12/2022.
//

import UIKit

class AButtonTemplate: UIButton {

	override init(frame: CGRect) {
		
		super.init(frame: frame);
		self.translatesAutoresizingMaskIntoConstraints = false;
		self.titleLabel?.font = UIFont.boldSystemFont(ofSize: 20.0);
		
		self.setTitleColor(__COLOR_WHITE, for: UIControl.State.normal);
		self.mSetBackgroundColor(__COLOR_PURPLE_DARK, for: UIControl.State.normal);
		
		self.setTitleColor(__COLOR_LEMON_LIGHT, for: UIControl.State.highlighted);
		self.mSetBackgroundColor(__COLOR_ORANGE_DARK, for: UIControl.State.highlighted);
	}
	
	required init?(coder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}
}
